src/linker/configuration.py

```diff
@@ -234,3 +220,100 @@ def _get_requests(environment: Dict[Any, Any], key: str) -> Dict[Any, Any]:
                         f"Assigning default value for {key} {default_key}: '{default_value}'"
                     )
         return requests
+
+    def _get_schema(self) -> Optional[PipelineSchema]:
+        """Validates the pipeline against supported schemas.
+
+        NOTE: this acts as the pipeline configurat file's validation method since
+        we can only find a matching schema if the file is valid.
+        """
+        errors = defaultdict(dict)
+        # Check that the pipeline specification contains a single "steps" outer key
+        if not len(self.pipeline) == 1 or not "steps" in self.pipeline:
+            errors[PIPELINE_ERRORS_KEY][
+                "generic"
+            ] = "The pipeline specification should contain a single 'steps' key."
+            exit_with_validation_error(dict(errors))
+
+        # Check that each of the pipeline steps also contains an implementation
+        metadata = load_yaml(IMPLEMENTATION_METADATA_PATH)
+        for step, step_config in self.pipeline["steps"].items():
+            if not "implementation" in step_config:
+                errors[PIPELINE_ERRORS_KEY][
+                    f"step {step}"
+                ] = "Does not contain an 'implementation'."
+            elif not "name" in step_config["implementation"]:
+                errors[PIPELINE_ERRORS_KEY][
+                    f"step {step}"
+                ] = "The implementation does not contain a 'name'."
+            elif not step_config["implementation"]["name"] in metadata:
+                errors[PIPELINE_ERRORS_KEY][
+                    f"step {step}"
+                ] = f"Implementation '{step_config['implementation']['name']}' is not defined in implementation_metadata.yaml."
```

Comment:
Is the user expected to know what `implementation_metadata.yaml` is?

Reply:
On, no I guess not. Maybe I just say that it's not a valid implementation name (and perhaps provide a list of the valid implementation names from the implementation_metadata.yaml)?

Reply:
Sure that makes sense